DDD-1.md

```diff
@@ -0,0 +1,173 @@
+# DDD-1: Support for multiple connector tasks for relational Debezium connectors
+
+## Motivation
+
+Currently, it is challenging to use Debezium Connector for SQL Server in a multi-tenanted environment where a single SQL Server instance hosts tens to hundreds of similar tenant-scoped databases due to the fact that a given connector instance can capture changes only from a single database. The challenges are:
+
+1. Kafka Connect resources. Deploying hundreds of connector instances will require a Kafka Connect cluster with significant CPU and memory resources.
+2. Cluster stability. If the resources above are insufficient (e.g. memory), it may cause a Connect worker crash which in turn will fail all connector tasks scheduled to the worker. In the best case scenario, those tasks can be just restarted. In a worse case scenario, the restarted connectors will produce duplicates since the source connectors currently don't implement exactly-once processing semantics. In an even worse case, a worker can fail during a long-running snapshot, handling which will require additional cleanup.
+3. SQL Server resources. Each server instance will have to maintain an extra set of persistent connections (one per database, tens to hundreds in total) which may require additional server resources.
```

Comment:
So this one wouldn't be mitigated by support for multiple tasks per-se, right? Rather via the ability to capture changes from multiple databases on one SQL Server host using one task.

Reply:
Correct. The support for multiple connectors per task will allow deploying a single connector per server instance and tuning its performance by changing `maxTasks` rather than deploying a different number of connectors and manually changing the list of databases for each.

Reply:
Ok, makes sense.